### ISSUE TITLE
Normalization (BN LN...)

### ISSUE THREAD

**Issue by sghong977** [sghong977/AIML_Basics#6]
관련자료
- [딥러닝 정규화?](https://realblack0.github.io/2020/03/29/normalization-standardization-regularization.html)

딥러닝 overfitting을 막는 방법으로 고전적으로 많이 언급되는 것이 batch normalization, weight regularization, dropout 등이 있다.  
이름이 정규화인게 3가지나 있어서 이거 공부할때 헷갈린다. Normalization, Standardization, Regularization.

가장 흔한 Batch Norm, 그리고 다른 normalization에 대해 보자.  

## BN
관련자료
- [BN 설명 참고](https://eehoeskrap.tistory.com/m/430)

### 개념  
- **batch norm**은 internal covariance shift 때문에 쓴다. 다음 레이어로 갈때 분포가 바뀔 수 있기 때문.
- 0 mean 1 std 분포로 노멀라이즈한다. 하지만 **input을 가지고 statistic**을 구하는 것 뿐만 아니라 **output에도 scaling, shift를 적용**함. (수식 봐주자) 얘네는 backprop으로 학습된다.  

### 쓰는 이유
- 쓰는 이유는 활성화 함수의 비선형성을 잃게 될 수 있어서, 이를 보전하기 위함이라함.
- 물론 요즘에는 shift를 해결해서라기보다는 다르게 설명하는듯 하다만...

###  train test모드 이야기
- data statistics 뽑아쓰는건 학습모드에서만 그렇다.
- 인퍼런스때에는 인풋 데이터의 mean std를 사용할수는 없는 노릇이니 deterministic하게 한다.
- 학습때 사용한 running mean, running variance를 불러와서 활용.

## 다른 normalization 기법  
관련자료
- [링크](https://subinium.github.io/introduction-to-normalization/)

아무래도 배치 사이즈가 작을 수 밖에 없는 task도 있고, recurrent한 구조는 bn을 적용하기가 뭐하다.  

![image](https://user-images.githubusercontent.com/46152199/186361920-4f979293-bd70-4f01-a2ca-fd61db5136bb.png)

- LN은 레이어에 하는데, 미니배치에서 feature dim이 같아야하는구나. 그래서 RNN같은거에 효과적이래.
- IN은 채널마다 따로해서, GAN style vector 얘기.
